android/src/main/java/com/reactnativechangeicon/ChangeIconModule.kt

```diff
@@ -30,15 +34,52 @@ class ChangeIconModule(reactContext: ReactApplicationContext, private val packag
         }
         promise.resolve(true)
         activity.packageManager.setComponentEnabledSetting(
-                ComponentName(packageName, activeClass),
-                PackageManager.COMPONENT_ENABLED_STATE_ENABLED,
-                PackageManager.DONT_KILL_APP
+            ComponentName(packageName, activeClass),
+            PackageManager.COMPONENT_ENABLED_STATE_ENABLED,
+            PackageManager.DONT_KILL_APP
         )
         activity.packageManager.setComponentEnabledSetting(
-                ComponentName(packageName, componentClass),
-                PackageManager.COMPONENT_ENABLED_STATE_DISABLED,
-                PackageManager.DONT_KILL_APP
+            ComponentName(packageName, componentClass),
+            PackageManager.COMPONENT_ENABLED_STATE_DISABLED,
+            PackageManager.DONT_KILL_APP
```

Comment:
@ensargunesdogdu, I think you missed some logic here, you need to remove this code snippet that disables current activity, and then set `iconChanged` to `true`.